test/extended/networking/bonds.go

```diff
@@ -0,0 +1,82 @@
+package networking
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	nadtypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"strings"
+)
+
+var _ = g.Describe("[sig-network][Feature:bond]", func() {
+	oc := exutil.NewCLI("bond")
+	f := oc.KubeFramework()
+
+	g.It("pod with bond interface should be created ", func() {
+		namespace := f.Namespace.Name
+		podName1, podName2 := "pod1", "pod2"
+		bondnad1, bondnad2 := "bondnad1", "bondnad2"
+
+		g.By("creating network attachment definitions")
+		err := createNetworkAttachmentDefinition(oc.AdminConfig(), namespace, "macvlannad", `{"cniVersion":"0.4.0","name":"macvlan-nad","plugins":[{ "type": "macvlan"}]}`)
+		o.Expect(err).NotTo(o.HaveOccurred(), "unable to create macvlan network-attachment-definition")
+		err = createBondNAD(oc.AdminConfig(), namespace, bondnad1, "10.0.0.10/24", "net1", "net2")
+		o.Expect(err).NotTo(o.HaveOccurred(), "unable to create bond network-attachment-definition")
+		err = createBondNAD(oc.AdminConfig(), namespace, bondnad2, "10.0.0.11/24", "net1", "net2")
+		o.Expect(err).NotTo(o.HaveOccurred(), "unable to create bond network-attachment-definition")
```

Comment:
These can be inlined:
```golang

		o.Expect( createNetworkAttachmentDefinition(oc.AdminConfig(), namespace, "macvlannad", `{"cniVersion":"0.4.0","name":"macvlan-nad","plugins":[{ "type": "macvlan"}]}`)).To(o.Succeed(), "unable to create macvlan network-attachment-definition")

		o.Expect(createBondNAD(oc.AdminConfig(), namespace, bondnad1, "10.0.0.10/24", "net1", "net2")).To(o.Succeed(), "unable to create bond network-attachment-definition")

		o.Expect(createBondNAD(oc.AdminConfig(), namespace, bondnad2, "10.0.0.11/24", "net1", "net2")).To(o.Succeed(), "unable to create bond network-attachment-definition")
```

Reply:
Done
although this make the lines too long for good readability

Reply:
> ```go
> o.Expect( createNetworkAttachmentDefinition(oc.AdminConfig(), namespace, "macvlannad", `{"cniVersion":"0.4.0","name":"macvlan-nad","plugins":[{ "type": "macvlan"}]}`)).To(o.Succeed(), "unable to create macvlan network-attachment-definition")
> ```

I'd have actually written 
```golang
o.Expect(
    createNetworkAttachmentDefinition(
        oc.AdminConfig(), 
        namespace, 
        "macvlannad", 
        `{"cniVersion":"0.4.0","name":"macvlan-nad","plugins":[{ "type": "macvlan"}]}`,),
).To(
    o.Succeed(), 
    "unable to create macvlan network-attachment-definition",
)
```
which addresses your concerns.

Reply:
Done